testlib/src/main/java/io/spine/test/TempDirectory.java

```diff
@@ -0,0 +1,300 @@
+        }
+        return extensionContext.getStore(NAMESPACE) //
+                               .getOrComputeIfAbsent(KEY,
+                                                     key -> tempDirProvider.get(parameterContext, extensionContext, TEMP_DIR_PREFIX),
+                                                     CloseablePath.class) //
+                               .get();
+    }
+
+    private static CloseablePath createDefaultTempDir(String dirPrefix) {
+        try {
+            return new CloseablePath(Files.createTempDirectory(dirPrefix));
+        }
+        catch (Exception ex) {
+            throw new ExtensionConfigurationException("Failed to create default temp directory", ex);
+        }
+    }
+
+    private static CloseablePath createCustomTempDir(ParentDirProvider parentDirProvider,
+                                                     ParameterContext parameterContext, ExtensionContext extensionContext, String dirPrefix) {
+        Path parentDir;
+        try {
+            parentDir = parentDirProvider.get(parameterContext, extensionContext);
+            requireNonNull(parentDir);
+        }
+        catch (Exception ex) {
+            throw new ParameterResolutionException("Failed to get parent directory from provider", ex);
+        }
+        try {
+            return new CloseablePath(Files.createTempDirectory(parentDir, dirPrefix));
+        }
+        catch (Exception ex) {
+            throw new ParameterResolutionException("Failed to create custom temp directory", ex);
+        }
+    }
+
+    private static class CloseablePath implements CloseableResource {
+
+        private final Path dir;
+
+        CloseablePath(Path dir) {
+            this.dir = dir;
+        }
+
+        Path get() {
+            return dir;
+        }
+
+        @Override
+        public void close() throws IOException {
+            Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+
+                @Override
+                public FileVisitResult visitFile(Path file, BasicFileAttributes attributes) throws IOException {
+                    return deleteAndContinue(file);
+                }
+
+                @Override
+                public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+                    return deleteAndContinue(dir);
+                }
+
+                private FileVisitResult deleteAndContinue(Path path) throws IOException {
+                    try {
+                        Files.delete(path);
+                    }
+                    catch (IOException ex) {
+                        throw new IOException(
+                                "Failed to delete temp directory " + dir.toAbsolutePath() + " at: " + path.toAbsolutePath(),
+                                ex);
+                    }
+                    return CONTINUE;
+                }
+            });
+        }
+    }
+}
```

Comment:
The EOF on this file is broken. See the GitHub complaints.

There is an `core.autocrlf` setting for Git clients to avoid this issue. Please have a read of the Git docs on this.